feat(stock): add option to view AMC calcs

jniles opened the pull request:
Adds the ability to view the AMC/CMM calculations in the Articles in Stock Registry.  A summary of the information is presented for the user to examine and the explicit formula used to calculate the AMC is shown for the user to validate.

Closes #5291.

**Example using Algorithm 1**
![image](https://user-images.githubusercontent.com/896472/106009293-764def00-60b8-11eb-815b-a9dc7ca6941f.png)

**Example using Algorithm 2**
![image](https://user-images.githubusercontent.com/896472/106009860-1ad03100-60b9-11eb-8927-78a1ac5fcb1e.png)

**Example using Algorithm 3**
![image](https://user-images.githubusercontent.com/896472/106008578-ba8cbf80-60b7-11eb-8c0f-be436068fd5a.png)

**Example using Algorithm 4 (MSH)**
![image](https://user-images.githubusercontent.com/896472/106008076-3f2b0e00-60b7-11eb-99d1-a9ae9078847f.png)

**Example showing Algorithm (MSH) for 12 months**
![image](https://user-images.githubusercontent.com/896472/106008960-1eaf8380-60b8-11eb-97fd-9ee13524dc84.png)

jniles committed:
feat(stock): add option to view AMC calcs

Adds the ability to view the AMC/CMM calculations in the Articles in
Stock Registry.  A summary of the information is presented for the user
to examine and the explicit formula used to calculate the AMC is shown
for the user to validate.

Closes #5291.

jmcameron reviewed (approved):
I looked over the code and it looked good.
I tested this with the IMCK database it looks great.   I did spot-check several of the CMM calculations to make sure the numbers worked.    Everything looks great.

One small/cosmetic/ignorable suggestion:  Make the values for the CMM column to links to the same function the action menu invokes.  It could include a tooltip:  "Click here to see the details of this calculation.".  Ignore this if you think this would slow things down too much.

mbayopanda commented on client/src/modules/stock/inventories/modals/amc.modal.js:
```diff
@@ -0,0 +1,31 @@
+      vm.isAlgo1 = vm.settings.average_consumption_algo === 'algo1';
+      vm.isAlgo2 = vm.settings.average_consumption_algo === 'algo2';
+      vm.isAlgo3 = vm.settings.average_consumption_algo === 'algo3';
+      vm.isAlgo4 = vm.settings.average_consumption_algo === 'algo_msh';
```
👍🏽 Clean and nice

mbayopanda reviewed (commented):
Good job @jniles 

I have noticed that : 

- The calculation of the Algo 3 doesn't match with what is displayed as an example **30.5 × (200 / 1) <> 32.97**: 
![image](https://user-images.githubusercontent.com/5445251/106120872-32f68d80-6157-11eb-99df-0b4d4d0a9fb7.png)

- The calculation of Algo 2 doesn't match with the example **30.5 × (200 / 185) <> 6100**
![image](https://user-images.githubusercontent.com/5445251/106121495-e95a7280-6157-11eb-962a-2a7bd5e12ddc.png)

It seems like the example for Algo 2 is given to Algo 3, and vice-versa

And the french translation

jniles committed:
feat(stock): double-click to view CMM calc

This commit adds the ability to double click on the CMM cell to view the
CMM calculation for the row as suggested by @jmcameron.

jmcameron reviewed (changes requested):
I could not get the double-clicking on the CMM value to work.  I suggest making it a link with tooltip, if you have time.

I went back and did some more spot-checks (which I should have done the first time!).   The calculations for Alg 1 and 4 seem fine.  But I'm seeing errors on 2 and 3.  See below.

![image](https://user-images.githubusercontent.com/62145/106193639-a18d1880-6162-11eb-9c0d-a0af66957295.png)

![image](https://user-images.githubusercontent.com/62145/106193734-c3869b00-6162-11eb-8761-6e683e6629e1.png)

jniles committed:
chore(stock): touch up AMC/CMM modal

Allows you to double click from anywhere in the cell, plus removes the
double click on the row header.  Starts process towards getting proper
AMC quantity in stock projections working.  Finally, adds translation
for french.

Closes #5291.

jmcameron reviewed (approved):
I tried the CMM double-click and it works fine now.  
Your decision to not add a tool-tip makes sense.
Let me know if you want me to review it again after straightening out the CMM calculations.

BTW, on the last screen-shot you have of the CMM calculations for Algo 4:  The fact that algorithms 1 and 2 have much bigger numbers than 3 and 4 is concerning too.
